ci/install-mdbook.sh

```diff
@@ -0,0 +1,19 @@
+set -euxo pipefail
+
+# Based on the Rust-Embedded WG's book CI
+# https://github.com/rust-embedded/book/blob/master/ci/install.sh
+
+main() {
+    # Note - this will only accept releases tagged with v0.3.x
```

Comment:
Not sure if we have semver expectations for the book, but this matches the previous behavior of `--version "^0.3"`